Source/Peripheral.swift

```diff
@@ -533,6 +533,32 @@ public class Peripheral {
     public func monitorWriteWithoutResponseReadiness() -> Observable<Void> {
         return delegateWrapper.rx_peripheralReadyToSendWriteWithoutResponse
     }
+    
+    /// Function that allow to open L2CAP channel for `Peripheral` instance.
+    /// For more information, please refer to
+    /// [What’s New in CoreBluetooth, 712, WWDC 2017](https://developer.apple.com/videos/play/wwdc2017/712/)
+    ///
+    /// - parameter PSM: `PSM` (Protocol/Service Multiplexer) of the channel
+    /// - returns: `Single` that emits `CBL2CAPChannel` when channel has opened
+    /// - since: iOS 11, tvOS 11, watchOS 4
+    #if os(iOS)
+        @available(iOS 11, tvOS 11, watchOS 4, *)
+        public func openL2CAPChannel(PSM: CBL2CAPPSM) -> Single<CBL2CAPChannel> {
+            let observable = delegateWrapper.rx_didOpenL2CAPChannel
```

Comment:
take(1)?

Reply:
Is `take(1)` really needed in this case? `flatmap` in next line emits `.just` or  throws an `error` so the stream will be closed anyway.

Reply:
If `rx_didOpenL2CAPChannel` emits **n** values, then the code in `flatMap` is called **n** times.
So I think `take(1)` is necessary or else `asSingle()` will emit an error.

Reply:
Yeah you guys right, will fix.